modules/apps/frontend-js/frontend-js-walkthrough-web/src/main/resources/META-INF/resources/Overlay.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {OverlayMask} from '@clayui/core';
+import React, {useCallback, useEffect, useState} from 'react';
+
+import {useObserveRect} from './useObserveRect';
+
+export function Overlay({popoverVisible, trigger}) {
+	const [overlayBounds, setOverlayBounds] = useState({
+		height: 0,
+		width: 0,
+		x: 0,
+		y: 0,
+	});
+
+	const updateOverlayBounds = useCallback(
+		(overlayBounds) => {
+			overlayBounds = overlayBounds
+				? overlayBounds
+				: trigger.getBoundingClientRect();
+
+			setOverlayBounds({
+				height: overlayBounds.height,
+				width: overlayBounds.width,
+				x: overlayBounds.x,
+				y: overlayBounds.y,
+			});
+		},
+		[trigger]
+	);
+
+	useEffect(() => {
+		if (trigger) {
+			updateOverlayBounds();
+		}
+
+		// eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Sorry, tonsillitis made me forget but why aren't we adding `trigger` to deps?

Reply:
I don't remember either, but I think it's safe to remove it and add the `trigger` as deps. Maybe there was another logic and we forgot to reanalyze it 🤷‍♂️.

Reply:
f3a731284c1817886c0c4c05a65ea3a5dc3dd3b3

This is why I love you, really